### PR TITLE
Making RDLI work in other environments.

### PR DESCRIPTION
**Pull request by TimSimpsonR**
- Changed code to get instance IP addresses.
- Made the sql query code work even when we have to ssh to another box
  to do it. (?!)
- Fixed some accounts API function tests to work with the "instance
  already exists" environment variable.
- Fixed a bug where the client used for a pre-created instance is not
  created using the right user.
- Changed a few places where we query MySql directly to be compatable
  with both local MySQL and the new Pexpect version.

### CODE REVIEW

File: tests/integration/tests/api/mgmt/accounts.py [hunk 1]
```diff
@@ -55,7 +56,8 @@ def test_invalid_account_fails(self):
     @test
     def test_account_zero_instances(self):
         account_info = self.client.accounts.show(self.user.tenant_id)
-        assert_equal(0, len(account_info.instances))
+        expected_instances = 0 if not existing_instance() else 1
```
Comment (ed-) [inline tests/integration/tests/api/mgmt/accounts.py:59]: Nice.

File: tests/integration/tests/util/__init__.py [hunk 2]
```diff
@@ -78,28 +77,18 @@ def assert_mysql_failure_msg_was_permissions_issue(msg):
...
     try:
-        engine.connect()
+        with mysql.create_mysql_connection(ip, user_name, password) as db:
+            pass
         fail("Should have failed to connect: mysql --host %s -u %s -p%s"
              % (ip, user_name, password))
-    except OperationalError as oe:
-        assert_mysql_failure_msg_was_permissions_issue(oe.message)
+    except mysql.MySqlPermissionsFailure:
+        return # Good, this is what we wanted.
```
Comment (ed-) [inline tests/integration/tests/util/__init__.py:88]: Expecting to fail. Failing if you fail to fail! Oh, what a tangled web you weave!

Also, not a big deal, but PEP8 and pep8 both want to see two spaces before the `#` of an inline comment

**Comment by ed-** [comment]
+1

File: tests/integration/tests/api/root.py [hunk 3]
```diff
@@ -80,8 +81,8 @@ def _root_local_sql(self):
         client = LocalSqlClient(engine)
         with client:
             t = text("""SELECT User, Host FROM mysql.user """
-                     """WHERE User=:user AND Host=:host;""")
-            result = client.execute(t, user=user, host=host)
+                     """WHERE User="%s" AND Host="%s" """ % user, host)
```
Comment (jcru) [inline tests/integration/tests/api/root.py]: Doesn't `user, host` require to be a tuple:

`% (user, host))`

Comment (ed-) [inline tests/integration/tests/api/root.py]: I'd think so, good catch.
sql.execute uses an older format where it takes the query with and then a series of args and plugs them in
But that format string should throw a TypeError as soon as we try it out, unless text() is handling that in a way I don't understand.

**Comment by hub-cap** [comment]
tim thats oppa gangnam style. +1

File: tests/integration/tests/util/mysql.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    def _init_engine(user, password, host):
+        return create_engine("mysql://%s:%s@%s:3306" % (user, password, host),
+                             pool_recycle=1800, echo=True)
+
+
+class PexpectMySqlConnection(object):
+
+    TIME_OUT = 30
+    NL = "\n\r"
+
```
Comment (cp16net) [inline tests/integration/tests/util/mysql.py]: doesnt look like this is used maybe it should be for the "\r\n" instead

File: tests/integration/tests/api/root.py (hunk 3 above)
Comment (TimSimpsonR) [inline tests/integration/tests/api/root.py]: I swear I ran this quite a few times. I'm not sure how it worked... I'm confused. Changing it anyway.

File: tests/integration/tests/util/mysql.py (hunk 4 above)
Comment (TimSimpsonR) [inline tests/integration/tests/util/mysql.py]: It has been eliminated!

**Comment by cp16net** [comment]
+1 :thumbsup:

**Comment by jcru** [comment]
+1

**Pull request merged**